core/src/main/scala/org/apache/spark/ui/jobs/JobPage.scala

```diff
@@ -337,7 +337,9 @@ private[ui] class JobPage(parent: JobsTab, store: AppStatusStore) extends WebUIP
       store.executorList(false), appStartTime)
 
     val operationGraphContent = store.asOption(store.operationGraphForJob(jobId)) match {
-      case Some(operationGraph) => UIUtils.showDagVizForJob(jobId, operationGraph)
+      case Some(operationGraph) => UIUtils.showDagVizForJob(jobId, operationGraph.sortWith(
+        _.rootCluster.id.replaceAll(RDDOperationGraph.STAGE_CLUSTER_PREFIX, "").toInt
```

Comment:
Add `getStageId` function in `RDDOperationGraph` to do this will be better.

Reply:
+1. `replaceAll` also seems like the wrong API to use, `substring` seems more correct.